{{cookiecutter.repo_name}}/README.rst

```diff
@@ -12,12 +12,118 @@ The XBlock SDK Workbench, including this XBlock, will be available on the list o
+use [edx-i18n-tools](https://github.com/edx/i18n-tools).
+
+After marking strings as translatable we have to create the raw message catalogs.
+These catalogs are created in `.po` files. For more information see
+[GNU PO file documentation](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html).
+These catalogs can be created by running:
+
+```shell
+$ make extract_translations
+```
+
+The previous command will create the necessary `.po` files under
+`{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/locale/en/LC_MESSAGES/text.po`.
+The `text.po` file is created from the `django-partial.po` file created by
+[`django-admin makemessages`](https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#message-files),
+this is why you will not see a `django-partial.po` file.
+
+### 3. Create language specific translations
+
+After creating the raw message catalogs, all translations should be filled out by the translator.
+Once translations are in place, use the next make target to compile the translation catalogs `.po` into
+`.mo` message files.
+
+```shell
+$ make compile_translations
+```
+
+The previous command will compile `.po` files using
+[`django-admin compilemessages`](https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#compiling-message-files).
+After compiling the `.po` file(s), `django-statici18n` is used to create language specific catalogs. See
+[`django-statici18n` documentation](https://django-statici18n.readthedocs.io/en/latest/) for more information.
+
+> **Note:** The `dev.run` make target will automatically compile any translations.
+
+> **Note:** To check if the source translation files (`.po`) are up-to-date run:
+> ```shell
+> $ make detect_changed_source_translations
+> ```
+
+### 4. Use `gettext` to translate strings
+
+Django will automatically use `gettext` and the compiled translations to translate strings.
+
+Troubleshooting
+----------------
+
+If there are any errors compiling `.po` files run the following command to validate your `.po` files:
+
+```shell
+$ make validate
+```
+
+See [django's i18n troubleshooting documentation](https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#troubleshooting-ugettext-incorrectly-detects-python-format-in-strings-with-percent-signs) for more information.
+
+Transifex
+-----------
 
-For more information on how to enable translations, visit the Open edX XBlock tutorial on Internationalization:
-http://edx.readthedocs.org/projects/xblock-tutorial/en/latest/edx_platform/edx_lms.html
+The `edx-i18n-tools` package facilitates retrieving and updating translations from and to [transfiex](https://transifex.com).
+See [config instructions](https://github.com/edx/i18n-tools#transifex-commands) for information on how to set up your
+transifex credentials.
 
-The template uses django-statici18n to provide translations to static javascript
-using `gettext`.
+To update transifex translations run:
+```shell
+$ make push_translations
+```
```

Comment:
Oh, I see you added the transifex commands here. Might as well move those to the workflow above.